auth/store.go

```diff
@@ -887,7 +864,7 @@ func (as *authStore) isAuthEnabled() bool {
 	return as.enabled
 }
 
-func NewAuthStore(be backend.Backend, indexWaiter func(uint64) <-chan struct{}) *authStore {
+func NewAuthStore(be backend.Backend, indexWaiter func(uint64) <-chan struct{}, tokenType, tokenOpts string) *authStore {
```

Comment:
Can this just be `func NewAuthStore(be backend.Backend, tokenProvider TokenProvider) *authStore`? There aren't any strict circular dependencies on `authStore` as far as I can tell. There can be a `NewTokenProvider(indexWaiter func(uint64) <-chan struct{}, tokenType, opts string) TokenProvider` (can probably fold the `opts` into `tokenType`) function to construct it.